packages/patternfly-3/patternfly-react-extensions/src/components/Select/Select.js

```diff
@@ -0,0 +1,176 @@
+import { Spinner, Button, ListGroup, ListGroupItem, TypeAheadSelect } from '../../../../patternfly-react/src/index';
+import { noop } from '../../common/helpers';
+import { createItemProps } from './SelectHelper';
+import SelectInput from './SelectInput';
+
+const Select = ({
+  cursor,
+  disabled,
+  emptyText,
+  focusOnMount,
+  isLoading,
+  isSearchable,
+  loadingText,
+  onItemClick,
+  onKeyDown,
+  onSearchChange,
+  onSearchClear,
+  onToggle,
+  open,
+  options,
+  placeholder,
+  searchValue,
+  selectedItem
+}) => {
+  const classes = classNames('select-container', { open });
+  const itemClasses = (itemclasses, cur) => classNames(itemclasses, { cursor: cur });
+
+  const renderResults = results =>
+    results.length === 0 ? (
+      <ListGroupItem id="select-empty" key="empty" className="select-empty-list">
+        <span id="empty-text">{emptyText}</span>
+      </ListGroupItem>
+    ) : (
+      results.map((opt, i) => (
+        <ListGroupItem
+          {...createItemProps(opt, selectedItem, itemClasses(opt.className || 'no-border', cursor === i), onItemClick)}
+        >
+          <EllipsisWithTooltip>
+            {searchValue && searchValue.length ? (
+              <TypeAheadSelect.Highlighter search={searchValue}>{opt.name}</TypeAheadSelect.Highlighter>
+            ) : (
+              opt.name
+            )}
+          </EllipsisWithTooltip>
+        </ListGroupItem>
+      ))
+    );
+
+  return (
+    <div className={classes}>
+      <Button disabled={disabled} onClick={onToggle} active={open} className="select-dropdown-toggle">
+        {selectedItem.id ? (
+          <EllipsisWithTooltip>{selectedItem.name}</EllipsisWithTooltip>
+        ) : (
+          <EllipsisWithTooltip>
+            <span className="select-toggle-placeholder">{placeholder}</span>
+          </EllipsisWithTooltip>
+        )}
+        <span className="caret select-caret" />
+      </Button>
+      {open && (
+        <div className="select-body-container">
+          {isSearchable && (
+            <SelectInput
+              onClear={onSearchClear}
+              focus={focusOnMount}
+              onChange={onSearchChange}
+              onKeyDown={onKeyDown}
+              placeholder={placeholder}
+              searchValue={searchValue}
+            />
+          )}
+          <ListGroup className="select-scrollable-list">
+            {isLoading ? (
+              <ListGroupItem id="select-loading" key="loading" className="select-loading-list">
+                <div id="select-loading-container">
+                  <Spinner id="select-spinner" loading size="sm" /> <span>{loadingText}</span>
+                </div>
+              </ListGroupItem>
+            ) : (
+              renderResults(options)
+            )}
+          </ListGroup>
+        </div>
+      )}
+    </div>
+  );
+};
+
+Select.propTypes = {
+  /** cursor position for keyboard navigation */
+  cursor: PropTypes.number,
```

Comment:
why are we controlling keyboard navigation via the arrow keys rather than allowing the native control (tab)?

Reply:
who said it has to be the arrow keys? its arrow keys just for the storybook example.

Reply:
ok, so what is the point in general?

Reply:
the arrow keys is a good example but it doesn't scroll if the cursor is going below the visible list. Maybe its best to remove this functionality for now and add it in a future PR, we want this hopefully to be merged today. Thoughts?

Reply:
I'd be ok with removing it and adding if/when its deemed necessary